### PR TITLE
#461 collapse post content

### PR DESCRIPTION
**Pull request by boscojwho**
# Checklist
- [x] I have read [CONTRIBUTING.md](./CONTRIBUTING.md)
- [x] I have described what this PR contains
- [x] This PR addresses one or more open issues that were assigned to me:
      - #461 
- [x] If this PR alters the UI, I have attached pictures/videos

# Pull Request Information

## About this Pull Request
Adds ability to minimize (collapse) a post's content when viewing as expanded post.
- Tested in light and dark modes.
- Tested in right-to-left language mode.
- Accessibility support.

## Screenshots and Videos
https://github.com/mlemgroup/mlem/assets/2549615/f0dbc7f5-b266-43ee-8932-70962ae255cb
https://github.com/mlemgroup/mlem/assets/2549615/df4b1a8e-1b3b-47d6-969e-707544e56d05

## Additional Context
Not so sure about the accessibility hint/label.

### CODE REVIEW

File: Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift [hunk 1]
```diff
@@ -23,66 +28,186 @@ struct LargePost: View {
...
+        }
+    }
+    @ViewBuilder
+    private var postBodyBackground: some View {
+        switch layoutMode {
+        case .minimize:
+            Color.secondarySystemBackground
+        default:
+            Color.clear
+        }
```
Comment (EricBAndrews) [inline Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift:66]: *nit*

2-case switch statements should be done as a simple if/else or ternary

File: Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift [hunk 2]
```diff
@@ -23,66 +28,186 @@ struct LargePost: View {
...
+    }
+    private var postBodyInsets: EdgeInsets {
+        switch layoutMode {
+        case .minimize:
+            return .init(top: 12, leading: 12, bottom: 12, trailing: 12)
+        default:
+            return .init(top: 0, leading: 0, bottom: 0, trailing: 0)
+        }
```
Comment (EricBAndrews) [inline Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift:74]: *nit*

See above re: 2-case switch

File: Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift [hunk 3]
```diff
@@ -23,66 +28,186 @@ struct LargePost: View {
...
+            .accessibilityHint("Expands post content.")
+    }
+    
+    private var postHeaderInsets: EdgeInsets {
+        switch layoutMode {
+        case .minimize:
+            return .init(top: 12, leading: 12, bottom: 12, trailing: 12)
+        default:
+            return .init(top: 0, leading: 0, bottom: 0, trailing: 0)
+        }
```
Comment (EricBAndrews) [inline Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift:148]: *nit*

2-case switch

File: Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift [hunk 4]
```diff
@@ -23,66 +28,186 @@ struct LargePost: View {
...
+    }
+    
+    @ViewBuilder
+    private var postHeaderBackground: some View {
+        switch layoutMode {
+        case .minimize:
+            Color.secondarySystemBackground
+        default:
+            Color.clear
```
Comment (EricBAndrews) [inline Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift:157]: *nit*

2-case switch

File: Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift [hunk 5]
```diff
@@ -23,66 +28,186 @@ struct LargePost: View {
...
+            .shadow(radius: 2)
+            .transition(
+                .scale(scale: 1.5)
+                .combined(with: .asymmetric(
+                    insertion: .push(from: .bottom),
+                    removal: .push(from: .top)))
+                .combined(with: .opacity)
+            )
+            .accessibilityLabel("Post content is minimized.")
+            .accessibilityHint("Expands post content.")
```
Comment (EricBAndrews) [inline Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift:139]: 🙏 kudos for accessibility tags

File: Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift [hunk 6]
```diff
@@ -23,66 +28,186 @@ struct LargePost: View {
...
-
+    private var lineLimit: Int? {
+        switch layoutMode {
+        case .maximize:
+            return nil
+        case .preferredSize:
+            return 8
+        case .minimize:
+            return 2
+        }
```
Comment (EricBAndrews) [inline Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift:57]: These can all be defined on the `LayoutMode` enum itself, then just used as `layoutMode.lineLimit` and `layoutMode.maxHeight`

**Review by EricBAndrews — CHANGES_REQUESTED** [review]
A couple nits and minor cleanliness/refactoring comments but looks good overall!

**Review by EricBAndrews — APPROVED** [review]

**Comment by fer0n** [comment]
@boscojwho I'm assuming it's not on purpose that the markdown in feed (large post size) is now no longer truncated?

**Comment by EricBAndrews** [comment]
I just noticed that as well, have a PR incoming to fix it

**Comment by boscojwho** [comment]
> @boscojwho I'm assuming it's not on purpose that the markdown in feed (large post size) is now no longer truncated?

Should be fixed in #501

**Comment by fer0n** [comment]
It's already fixed, yes. I also made some changes to the markdown preview which would've also taken care of it anyways :)

**Pull request merged**